MIDIchlorians/MIDIchlorians/ViewController.swift

```diff
@@ -91,13 +111,13 @@ class ViewController: UIViewController {
 
     // Saves the current session
     func saveCurrentSession() {
-        currentSession = dataManager.saveSession(
-            currentSession.getSessionName() ?? Config.DefaultSessionName, currentSession)
+        let sessionName = currentSession.getSessionName() ?? Config.DefaultSessionName
+        currentSession = dataManager.saveSession(sessionName, currentSession)
         sessionTableViewController.sessions = dataManager.loadAllSessionNames()
     }
 
     // Tries to load a session, if no sessions exists then returns nil
-    private func loadFirstSessionIfExsists() -> Session? {
+    private func loadFirstSessionIfExists() -> Session? {
         func loadTillSuccessOrEnd(names: [String]) -> Session? {
```

Comment:
Would prefer an iterative method (unless recursive is the only way). Not a significant issue though.